plugins/DataLoadRlog/rlog_parser.cpp

```diff
@@ -78,3 +92,23 @@ bool RlogMessageParser::parseMessageImpl(const std::string& topic_name, capnp::D
   }
   return true;
 }
+
+bool RlogMessageParser::parseCanMessage(
+  const std::string& topic_name, capnp::DynamicList::Reader listValue, double time_stamp) 
+{
+  for(auto elem : listValue) {
+    auto value = elem.as<capnp::DynamicStruct>();
+    uint8_t bus = value.get("src").as<uint8_t>();
+    if (bus == 1) {
+      if (parser != nullptr && packer != nullptr) {
+        parser->UpdateCans((uint64_t)(time_stamp), value);
+        for (auto& sg : parser->query_latest()) {
```

Comment:
You can do a single call to query_latest after you parsed all the values from the DynamicList